### DISCUSSION TITLE
"Always use early returns" - LinkedIn Edition

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#124] [General]
# "Always use early returns" - LinkedIn Edition

Ahhhh dear LinkedIn - a pool of gems where everyone is expert in everything. Over the time I collected some trophies from there and today I want to discuss one of them: "Always use early returns".

https://steven-giesel.com/blogPost/ba94e3fe-57e7-4f27-83a7-97f349a4b39c

**Comment by jslicer** [comment]
That original post misses one more good coding standard: never return `null` when the return type is an `IEnumerable` or descendant - return an empty version of that so callers don't have to constantly check for `null`.

**Reply by linkdotnet** [reply]
I wouldn't say *never* return `null`<sup>*</sup> but I get your sentiment and I totally agree.

<sup>*</sup> You might run into cases where there is a useful distinction between returning `null` or an empty list. That includes performance optimization (the dotnet team does use this pattern quite often here and there). That might also include a business need where you have to differ between null and empty. 
But of course, we are not talking the 80% case here.
